internal/node/node.go

```diff
@@ -13,11 +14,12 @@ import (
 //go:generate mockgen -source=node.go -package=node -destination=mock_node.go
 
 type Node interface {
-	IsNodeSchedulable(node *v1.Node) bool
 	GetNodesListBySelector(ctx context.Context, selector map[string]string) ([]v1.Node, error)
 	GetNumTargetedNodes(ctx context.Context, selector map[string]string) (int, error)
-	UpdateLabels(ctx context.Context, node *v1.Node, toBeAdded, toBeRemoved []string) error
+	IsNodeSchedulable(node *v1.Node) bool
 	NodeBecomeReadyAfter(node *v1.Node, checkTime metav1.Time) bool
+	RemoveAllKmodLabels(ctx context.Context, node *v1.Node) error
```

Comment:
```suggestion
	RemoveKmodReadyLabels(ctx context.Context, node *v1.Node) error
```

Reply:
i would suggest renaming it "RemoveNodeReadyLabels", since it is not related to a specific kmod, but to any kmod on the node

Reply:
> i would suggest renaming it "RemoveNodeReadyLabels", since it is not related to a specific kmod, but to any kmod on the node

`RemoveNodeReadyLabels` is not to general? I think for example a node can have label for his Ready status.

Reply:
ok, then RemoveNodeKmodReadyLabels. @ybettan wdyt?

Reply:
@ybettan

Reply:
I would go with `RemoveNodeReadyLabels`.

There is no "ready" label on the node by default:
```
$ kubectl get node/minikube -o yaml | yq '.metadata.labels'
beta.kubernetes.io/arch: amd64
beta.kubernetes.io/os: linux
kubernetes.io/arch: amd64
kubernetes.io/hostname: minikube
kubernetes.io/os: linux
minikube.k8s.io/commit: 5883c09216182566a63dff4c326a6fc9ed2982ff
minikube.k8s.io/name: minikube
minikube.k8s.io/primary: "true"
minikube.k8s.io/updated_at: 2024_11_25T11_06_54_0700
minikube.k8s.io/version: v1.33.1
node-role.kubernetes.io/control-plane: ""
node.kubernetes.io/exclude-from-external-load-balancers: ""
```

The "ready" on the node is a `condition` and not a label:
```
$ kubectl get node/minikube -o yaml | yq '.status.conditions[] | select (.type == "Ready")'
lastHeartbeatTime: "2024-11-25T09:07:08Z"
lastTransitionTime: "2024-11-25T09:07:08Z"
message: kubelet is posting ready status
reason: KubeletReady
status: "True"
type: Ready
```

Reply:
You are right, I changed it to `RemoveNodeReadyLabels` then.